### ISSUE TITLE
Duplication in hierarchical dimensions

### ISSUE THREAD

**Issue by kronmar** [visualize-admin/visualization-tool#1020]
Concerns test.visualize.admin.ch, with data source INT. For reproduction, look at cube with identifier `Jstat_fullHierarchy_C-94` (name is "NFI: All target values by ownership" from the 06.04.2023.

After some internal feedback, we decided to try to add the grandchildren directly to the root. I.e. we have four hierarchies on the dimension `nfi:unitOfReference` with Root <https://ld.admin.ch/country/CHE>:

- CH - Cantons
- CH - Production Regions - Economic Regions
- CH - Protection Forest Regions - Economic Regions
- CH - Economics Regions **this one is new**

They all use the path `schema:containsPlace` and a `sh:targetClass` appropriate for the corresponding hierarchical branch.

In the drop-down menu, children and grandchildren of the hierarchy are displayed:
![image](https://user-images.githubusercontent.com/95312586/230377138-ee621f26-2ecb-46bf-86d4-3d79daf43849.png)

I don't think this is a data issue, as [this](https://s.zazuko.com/2eP7eHY) should demonstrate.

**Comment by Rdataflow** [comment]
👀 looking at the data published by @kronmar this seems really odd parsing in a very non-conformant way according to the cube spec. (i.e. including some non targetClasses)

potentially related to inference i.e. due to parsing other cubes shapes?

**Comment by kronmar** [comment]
Yes, you are correct. The `https://ld.admin.ch/dimension/bgdi/biota/economicregions/xx` are indeed both children and siblings to the `https://ld.admin.ch/dimension/bgdi/biota/productionregions/xx`. 

Another find that might be unrelated is this: 
![image](https://user-images.githubusercontent.com/95312586/232420752-1f09d83f-7102-4931-9a8f-63b40a40a094.png)

You can find it [here](https://test.visualize.admin.ch/en/v/I1StHhL43GU2). You'll see that the root is mentioned four times (number of branches in the hierarchy?). Again, not really sure if this is related or not. If you wish, I'll open a separate issue.
